main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala

```diff
@@ -163,25 +172,25 @@ object CompletionProvider {
     // We check type and effect first because for example following def we do not want completions other than type and effect if applicable.
     context.prefix match {
       case channelKeywordRegex() | doubleColonRegex() | tripleColonRegex() => getExpCompletions()
-      case withRegex() => WithCompleter.getCompletions(context) map (withComp => withComp.toCompletionItem(context))
```

Comment:
It would be useful if these refactorings (CompletionItem to Completion) could happen in a separate PR.